action-core/src/main/java/com/adyen/checkout/action/core/internal/ui/DefaultGenericActionDelegate.kt

```diff
@@ -101,35 +103,41 @@ internal class DefaultGenericActionDelegate(
     }
 
     override fun handleAction(action: Action, activity: Activity) {
+        savedStateHandle[ACTION_KEY] = action
+
         // This check is to support an older flow where you might need to call handleAction several times with 3DS2.
         // Initially handleAction is called with a fingerprint action then with a challenge action.
         // During this whole flow the same transaction instance should be used for both fingerprint and challenge.
         // Therefore we are making sure the same delegate persists when handleAction is called again.
         if (isOld3DS2Flow(action)) {
             adyenLog(AdyenLogLevel.DEBUG) { "Continuing the handling of 3ds2 challenge with old flow." }
         } else {
-            val delegate = actionDelegateProvider.getDelegate(
-                action = action,
-                checkoutConfiguration = checkoutConfiguration,
-                savedStateHandle = savedStateHandle,
-                application = activity.application,
-            )
-            this._delegate = delegate
-            adyenLog(AdyenLogLevel.DEBUG) { "Created delegate of type ${delegate::class.simpleName}" }
-
-            if (delegate is RedirectableDelegate) {
-                onRedirectListener?.let { delegate.setOnRedirectListener(it) }
-            }
+            createDelegateAndObserve(action, activity.application)
+        }
 
-            delegate.initialize(coroutineScope)
+        delegate.handleAction(action, activity)
+    }
 
-            observeDetails(delegate)
-            observeExceptions(delegate)
-            observePermissionRequests(delegate)
-            observeViewFlow(delegate)
+    private fun createDelegateAndObserve(action: Action, application: Application) {
```

Comment:
we can separate it into two functions for creating and observing.

Reply:
Let's say we have those 2 methods, would you then use it like this:
```Kotlin
fun someFun() {
    createDelegate(action, activity.application)
    observeDelegate(_delegate)
}
```
or call it from the `createDelegate` method?:
```Kotlin
fun createDelegate(action: Action, application: Application) {
    val delegate = ...
    observeDelegate(delegate)
}
```

Reply:
not a strong preference but second one seems clearer to me. the idea behind it was the function does 2 separate things and could be more granular.